pkg/handlers/pptasapi/pptas_report.go

```diff
@@ -21,12 +23,23 @@ type PPTASReportsHandler struct {
 func (h PPTASReportsHandler) Handle(params pptasop.PptasReportsParams) middleware.Responder {
 	return h.AuditableAppContextFromRequestWithErrors(params.HTTPRequest,
 		func(appCtx appcontext.AppContext) (middleware.Responder, error) {
-			var searchParams services.MoveTaskOrderFetcherParams
+			var searchParams services.MovesForPPTASFetcherParams
 			if params.Since != nil {
 				since := handlers.FmtDateTimePtrToPop(params.Since)
 				searchParams.Since = &since
 			}
 
+			clientCert := authentication.ClientCertFromContext(appCtx.HTTPRequest().Context())
+
+			if clientCert.PPTASAffiliation != nil {
+				if *searchParams.Affiliation != models.AffiliationNAVY.String() && *searchParams.Affiliation != models.AffiliationMARINES.String() {
+					appCtx.Logger().Error("Invalid branch provided for filtering reports", zap.String("branch", *searchParams.Affiliation))
+					return pptasop.NewPptasReportsInternalServerError().WithPayload(payloads.InternalServerError(nil, h.GetTraceIDFromRequest(params.HTTPRequest))), nil
+				}
+			}
+
+			searchParams.Affiliation = (*string)(clientCert.PPTASAffiliation)
```

Comment:
```suggestion
			clientCert := authentication.ClientCertFromContext(appCtx.HTTPRequest().Context())

			if clientCert.PPTASAffiliation != nil {
				searchParams.Affiliation = (*string)(clientCert.PPTASAffiliation)
				if *searchParams.Affiliation != models.AffiliationNAVY.String() && *searchParams.Affiliation != models.AffiliationMARINES.String() {
					appCtx.Logger().Error("Invalid branch provided for filtering reports", zap.String("branch", *searchParams.Affiliation))
					return pptasop.NewPptasReportsInternalServerError().WithPayload(payloads.InternalServerError(nil, h.GetTraceIDFromRequest(params.HTTPRequest))), nil
				}
			}
```

nil pointer error on `*searchParams.Affiliation` as it's not set yet.

Reply:
Thank you for pointing this out, and extra big thanks for just fixing it outright. The 🐐